operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/CachingEventSource.java

```diff
@@ -69,4 +67,9 @@ public void stop() throws OperatorException {
     super.stop();
     cache.close();
   }
+
+  @Override
+  public T getAssociated(P primary) {
```

Comment:
Why it needs the whole resource? I think this should be ideally as simple as possible API and just use the ResourceID

Reply:
It's just generalizing the method that's already present on `InformerEventSource`. This method will always be called in a context where the primary is available anyway so why restrict it to only providing the `ResourceID`?

Reply:
not sure how we agreed on this, I still think it should be ResourceID :)